File: app/src/main/java/yuva/assignment/wiproapplication/app/MyApplication.java
package yuva.assignment.wiproapplication.app;

import android.app.Application;

import yuva.assignment.wiproapplication.diprovider.ApiModule;
import yuva.assignment.wiproapplication.diprovider.AppModule;

public class MyApplication extends Application {

    private static ApiComponent mApiComponent;
    private static MyApplication mInstance;

    public static synchronized MyApplication getInstance() {
        return mInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
        mApiComponent = DaggerApiComponent.builder()
                .appModule(new AppModule(this))
                .apiModule(new ApiModule(Constant.BASE_URL))
                .build();

    }

    public static ApiComponent getNetComponent() {
        return mApiComponent;
    }
}

File: app/src/main/java/yuva/assignment/wiproapplication/app/MyApplication.kt
package yuva.assignment.wiproapplication.app

import android.app.Application

import yuva.assignment.wiproapplication.diprovider.ApiModule
import yuva.assignment.wiproapplication.diprovider.AppModule

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        netComponent = DaggerApiComponent.builder()
                .appModule(AppModule(this))
                .apiModule(ApiModule(Constant.BASE_URL))
                .build()
    }

    companion object {
        var netComponent: ApiComponent? = null
            private set
        @get:Synchronized
        var instance: MyApplication? = null
            private set
    }
}